CLDC-3405-update-retirement-age-validation

arfacamble opened the pull request:
update retirement soft validation pages and copy in en.yml to reflect changes in english retirement age

https://dluhcdigital.atlassian.net/browse/CLDC-3405

arfacamble committed:
update retirement soft validation pages and copy in en.yml to reflect changes in english retirement age

RachaelBooth commented on spec/models/lettings_log_spec.rb:
```diff
@@ -3382,64 +3382,6 @@ def previous_postcode_fields
-      end
-    end
-
-    context "when a postcode contains unicode characters" do
```
This looks like an unrelated test that shouldn't have been in this context block - we should probably keep it (but in a better place) unless it's already a duplicate.

arfacamble replied:
good spot!
what a strange place to put that test!

RachaelBooth commented on app/models/log.rb:
```diff
@@ -205,10 +205,6 @@ def blank_invalid_non_setup_fields!
-      retirement_age_for_person(person_num)
-    end
-
     define_method("plural_gender_for_person_#{person_num}") do
```
Does this get used anywhere else?

arfacamble replied:
I don't believe so, but it's always impossible to tell for sure with rails since someone _can_ do something very strange like `obj.send(%w(retirement age for person 1).join("_"))`
but I've global searched retirement_age_for which now only gets hits in the old form json files and I can't think of any other way it might be getting called that isn't far fetched.
The tests do all pass as well, so you would hope that that covers things?

RachaelBooth replied:
Ah, rails... That implies we probably could remove it but it seems right to be wary.

arfacamble committed:
update soft validation tests to display interruption screens under the correct conditions given changes to state retirement age
reduction of variation in retirement age means lots of code related to determining the retirement age given an individuals gender can also be removed
testing updated
